test/e2e/common.go

```diff
@@ -748,3 +751,81 @@ func (Metal3LogCollector) CollectMachinePoolLog(_ context.Context, _ client.Clie
 func (Metal3LogCollector) CollectInfrastructureLogs(_ context.Context, _ client.Client, _ *clusterv1.Cluster, _ string) error {
 	return fmt.Errorf("CollectInfrastructureLogs not implemented")
 }
+
+// semVer = v1.2.0
+// GitCheckout checkouts the repo that is at path repoPath to branch checkoutBranch.
+func GitCheckout(repoPath string, checkoutBranch string) error {
```

Comment:
Could we handle this using something more go-native, like https://github.com/go-git/go-git?